src/model_desc/paths_desc.ml

```diff
@@ -60,95 +60,96 @@ module General_paths = struct
 
   let rec identifier : Paths.Identifier.t t =
     Variant
```

Comment:
There's an opportunity to have a shorter output, this could return the key.

Reply:
True, though I wasn't thinking about printing the keys when I made them so they might not be particularly legible. In particular they're backward, so `Foo.bar` is `v_bar.m_Foo` IIRC (I thought many identifiers would have the same prefix, so reversing them might make it quicker to reject when comparing them)

Reply:
This would have no effect for `hash` and `equal`, only `compare`, which is used less.